### DISCUSSION TITLE
Java 17 will be the minimum version for Quarkus 3.7

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#37416] [Quarkus Blog/Website]
# Java 17 will be the minimum version for Quarkus 3.7

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/java-17/

**Comment by smsajid** [comment]
This is a welcome move. I am already into Java 21

**Reply by gsmet** [reply]
Note that this only impacts our ability in the Quarkus project itself to use Java 17 features.
You can already use Java 21 with Quarkus 3.6 in your applications.

**Comment by bulldog98** [comment]
This is sensible, at the latest when dependencies require Java 17, you would have to move. But I know from own experience that there are still programs out there that run with java 8 and will most likely never be updated.

**Reply by gsmet** [reply]
Well, if they are not updated, they would have to stay on earlier versions.

For Java 8, it would be an unmaintained version so I wouldn't recommend it but for Java 11, they can use 3.2 LTS.

**Reply by bulldog98** [reply]
The Java 8 example was outside of the context of quarkus, but yes there are unmaintained versions of libraries involved.

**Comment by olekstomek** [comment]
It's a waste of time looking back at old technologies. ;)

**Comment by wclaeys** [comment]
Will Quarkus 3.7 be a new LTS?

**Reply by serrodcal** [reply]
I'm also interesting on it. Because we are working on a project, we recently migrated to Java21 and we want to set a LTS version. Thanks!

**Reply by gsmet** [reply]
Sorry for the delay, we were working on preparing a proper answer to your question. Here it is: https://quarkus.io/blog/our-plans-for-quarkus-3-7-3-8-3-9-released/ .

**Comment by fugerit79** [comment]
> https://quarkus.io/blog/java-17/

Hello and thanks with the great work with Quarkus. I'm using it on many projects.
While it is perfectly reasonable to move from java 11 to java 17, wouldn't it be a good reason to increase quarkus major version?
(from 3 to 4?)

Thanks again!

**Reply by maxandersen** [reply]
No. We already deprecated Java 11 at Quarkus 3.0 to signal it would go away and introduced LTS to provide options.
